modules/telestion-api/src/main/java/de/wuespace/telestion/api/verticle/TelestionVerticle.java

```diff
@@ -0,0 +1,201 @@
+package de.wuespace.telestion.api.verticle;
+
+import io.vertx.core.AbstractVerticle;
+import io.vertx.core.Promise;
+import io.vertx.core.json.JsonObject;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.lang.reflect.ParameterizedType;
+
+/**
+ * An abstract verticle class that you can extend to write your own verticle classes.
+ * <p>
+ * It's extends the {@link AbstractVerticle} from Vert.x to add support for default configurations
+ * and type-safe usage of the configuration JSON object provided by Vert.x.
+ *
+ * @param <T> the type of your Configuration class
+ * @author Cedric Bös, Pablo Klaschka, Jan von Pichowski, Ludwig Richter
+ */
+public abstract class TelestionVerticle<T extends TelestionConfiguration> extends AbstractVerticle {
+	/**
+	 * The default verticle configuration in a generic format.
+	 */
+	private JsonObject defaultGenericConfig = new JsonObject();
+	/**
+	 * The default verticle configuration in the Configuration type format.<p>
+	 * Is <code>null</code> when no type via {@link #getConfigType()} is given.
+	 */
+	private T defaultConfig;
+
+	/**
+	 * The verticle configuration in a generic format.
+	 */
+	private JsonObject genericConfig = new JsonObject();
+	/**
+	 * The verticle configuration in the Configuration type format.<p>
+	 * Is <code>null</code> when no type via {@link #getConfigType()} is given.
+	 */
+	private T config;
+
+	/**
+	 * The default logger instance.
+	 */
+	protected final Logger logger = LoggerFactory.getLogger(getClass());
+
+	/**
+	 * Get the Configuration Class type from the inheriting class.<p>
+	 * Overwrite this method and return the class type of your Configuration
+	 * to allow usage of the {@link #getConfig()} and {@link #getDefaultConfig()} methods.
+	 * @return the Configuration Class type
```

Comment:
```suggestion
	 * Get the Configuration Class type from the inheriting class.
	 * @return the Configuration Class type
```

Remove outdated documentation